### PR TITLE
Added multi agent tests

### PR DESCRIPTION
**Pull request by cszsol**
https://bbpteam.epfl.ch/project/issues/browse/BBPP164-56

### CODE REVIEW

File: tests/multi_agents/test_supervisor_multi_agent.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+from src.neuroagent.multi_agents import SupervisorMultiAgent
+
+
+def test_create_main_agent():
+    mock_llm = MagicMock()
+    bind_function_result = MagicMock()
+    bind_function_result.__ror__.return_value = {}
+    mock_llm.bind_functions.return_value = bind_function_result
+    data = {"llm": mock_llm, "agents": [("agent1",)]}
+    from src.neuroagent.multi_agents import SupervisorMultiAgent
```
Comment (KeremKurban) [inline tests/multi_agents/test_supervisor_multi_agent.py:17]: why are we importing SupervisorMultiAgent here again ?

File: tests/multi_agents/test_supervisor_multi_agent.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        name="test_agent",
+    )
+
+    async def mock_ainvoke(_):
+        return {"messages": [mock_message]}
+
+    agent_state = MagicMock()
+    agent = MagicMock()
+    agent.ainvoke = mock_ainvoke
+    from src.neuroagent.multi_agents import SupervisorMultiAgent
```
Comment (KeremKurban) [inline tests/multi_agents/test_supervisor_multi_agent.py:37]: same as above

File: tests/multi_agents/test_supervisor_multi_agent.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    bind_function_result = MagicMock()
+    bind_function_result.__ror__.return_value = {}
+    mock_llm.bind_functions.return_value = bind_function_result
+    data = {"llm": mock_llm, "agents": [("agent1",)]}
+    from src.neuroagent.multi_agents import SupervisorMultiAgent
+
+    result = SupervisorMultiAgent.create_main_agent(data)
+    assert "main_agent" in result
+    assert "summarizer" in result
+
```
Comment (KeremKurban) [inline tests/multi_agents/test_supervisor_multi_agent.py:22]: this is rather test_create_main_agent_initialization. 
i was going to suggest to add assert agent1 in agents but i see it in test_create_graph.

File: tests/agents/test_simple_chat_agent.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
 from neuroagent.agents import AgentOutput, AgentStep, SimpleChatAgent
 
 
+@pytest.mark.httpx_mock(can_send_already_matched_responses=True)
```
Comment (KeremKurban) [inline tests/agents/test_simple_chat_agent.py:12]: just wondering. is this for the cases where we have sent the exact request before somewhere else in the tests?

**Review by KeremKurban — COMMENTED** [review]
Minor comments and suggestions

File: tests/multi_agents/test_supervisor_multi_agent.py (hunk 1 above)
Comment (cszsol) [reply tests/multi_agents/test_supervisor_multi_agent.py:17]: It was left from a previous version and is no longer needed. I deleted it.

File: tests/multi_agents/test_supervisor_multi_agent.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    mock_message = HumanMessage(
+        content="hello",
+        name="test_agent",
+    )
+
+    async def mock_ainvoke(_):
+        return {"messages": [mock_message]}
+
+    agent_state = MagicMock()
+    agent = MagicMock()
```
Comment (WonderPG) [inline tests/multi_agents/test_supervisor_multi_agent.py:34]: If a dunder method is not mocked, magic mock can be replaced with a regular mock I believe

**Review by WonderPG — COMMENTED** [review]
Thanks for the MR !

File: tests/multi_agents/test_supervisor_multi_agent.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    assert result["messages"][0].content == "hello"
+
+
+def test_create_graph():
+    class FakeChatModel(GenericFakeChatModel):
+        def bind_tools(self, functions: list):
+            return self
+
+        def bind_functions(self, **kwargs):
+            return self
```
[the remark this answers is not in the document]
Comment (cszsol) [reply tests/multi_agents/test_supervisor_multi_agent.py:88]: I changed your fixture slightly and now I was able to use it.

**Review by WonderPG — APPROVED** [review]

**Pull request merged**